ebos/ecltransmissibility.hh

```diff
@@ -149,17 +150,65 @@ public:
         for (unsigned dimIdx = 0; dimIdx < dimWorld; ++dimIdx)
             axisCentroids[dimIdx].resize(numElements);
 
+        std::vector<double> centroids;
+#if HAVE_MPI
+        size_t cells = vanguard_.grid().numCells();
+        const auto& comm = vanguard_.gridView().comm();
+        if (global && comm.size() > 1) {
+            if (comm.rank() == 0) {
+                std::vector<size_t> sizes(comm.size());
+                comm.gather(&cells, sizes.data(), 1, 0);
+                for (int i = 1; i < comm.size(); ++i) {
+                    std::vector<int> cell_id(sizes[i]);
+                    MPI_Recv(cell_id.data(), sizes[i], MPI_INT,
+                             i, 0, MPI_COMM_WORLD, MPI_STATUS_IGNORE);
+                    centroids.resize(dimWorld * sizes[i]);
+
+                    auto cIt = centroids.begin();
+                    for (int idx : cell_id) {
+                        const auto& centroid = eclGrid.getCellCenter(idx);
+                        for (const auto& it : centroid)
+                            *cIt++ = it;
+                    }
+                    MPI_Send(centroids.data(), dimWorld * sizes[i],
+                             MPI_DOUBLE, i, 0, MPI_COMM_WORLD);
+                }
+                centroids.clear();
+            } else {
+                comm.gather(&cells, &cells, 1, 0);
```

Comment:
This looks weired (at least to me). I know the second parameter is not used for writing on this process, but it causes confusion.
```
std::vector<size_t> sizes;
comm.gather(&cells, sizes.data(), 1, 0);
```
Or even better move it out of the if-else.

Reply:
uhm, okay, we get confused by different things, but done as you request (moved the vector out).